From d30ff06653751be0d94938355c8842ddbb3c868a Mon Sep 17 00:00:00 2001
From: Matthias Maennich <maennich@google.com>
Date: Wed, 25 Mar 2020 21:21:54 +0100
Subject: [PATCH 664/830] ANDROID: gki_defconfig: enabled UAPI header compile
 testing
X-NVConfidentiality: public

To ensure UAPI headers will compile standalone and can (later) be
monitored for API/ABI stability, enable the feature accordingly.

That seams like an additional overhead at compile time, but runs
smoothly while we are waiting for the LTO link to finish :-)

Bug: 73298553
Signed-off-by: Matthias Maennich <maennich@google.com>
Change-Id: Iae4f67f430ee2a3000f9f035aa0cd4392e72460f
---
 arch/arm64/configs/gki_defconfig | 2 ++
 arch/x86/configs/gki_defconfig   | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 7e87da00513c..c9516e76d5c9 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -1,3 +1,4 @@
+CONFIG_UAPI_HEADER_TEST=y
 CONFIG_LOCALVERSION="-mainline"
 CONFIG_AUDIT=y
 CONFIG_NO_HZ=y
@@ -471,6 +472,7 @@ CONFIG_DMA_CMA=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_INFO=y
 # CONFIG_ENABLE_MUST_CHECK is not set
+CONFIG_HEADERS_INSTALL=y
 # CONFIG_SECTION_MISMATCH_WARN_ONLY is not set
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_DEBUG_STACK_USAGE=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index 98d5b7196f6a..ca992f7ec2b9 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -1,3 +1,4 @@
+CONFIG_UAPI_HEADER_TEST=y
 CONFIG_LOCALVERSION="-mainline"
 # CONFIG_USELIB is not set
 CONFIG_AUDIT=y
@@ -414,6 +415,7 @@ CONFIG_DMA_CMA=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_INFO=y
 # CONFIG_ENABLE_MUST_CHECK is not set
+CONFIG_HEADERS_INSTALL=y
 # CONFIG_SECTION_MISMATCH_WARN_ONLY is not set
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_DEBUG_STACK_USAGE=y
-- 
2.17.1

